src/shared/frame-error-capture.js

```diff
@@ -41,12 +79,28 @@ export function sendError(error, context) {
     return;
   }
 
-  const data = { type: 'hypothesis-error', error, context };
+  const data = {
+    type: 'hypothesis-error',
+    error: error instanceof Error ? error : serializeError(error),
+    context,
+  };
+
   try {
-    // Try to send the error. This will currently fail in browsers which don't
-    // support structured cloning of exceptions. For these we'll need to implement
-    // a fallback.
-    errorDestination.postMessage(data, '*');
+    // Try to send the error. If this fails because the browser doesn't support
+    // structured cloning of errors, use a fallback.
+    try {
+      errorDestination.postMessage(data, '*');
+    } catch (postErr) {
+      if (
+        postErr instanceof DOMException &&
+        postErr.name === 'DataCloneError'
```

Comment:
I guess you have tested this in the browsers that don't currently support serialisation of `Error`.

Reply:
Yes. This can be tested from the browser console like so:

```
> try { window.postMessage(new Error('test'), '*'); } catch (err) { lastErr = err };
< DataCloneError: The object can not be cloned.
> lastErr instanceof DOMException
< true
> lastErr.name
< "DataCloneError"
```